### ISSUE TITLE
Update available environment variables so the user does not need to restart their terminal

### ISSUE THREAD

**Issue by lloydjatkinson** [microsoft/winget-cli#3077]
### Description of the new feature / enhancement

Some software that is installed adds itself to the `PATH` environment variable. These changes are not surfaced to the current terminal session and requires the user to close the terminal and open it again. I consider this to be a huge blocker for creating fully automated software installations as the environment variables it added are often needed in the next step.

Consider NVM as per the docs here: [https://learn.microsoft.com/en-us/windows/dev-environment/javascript/nodejs-on-windows](https://learn.microsoft.com/en-us/windows/dev-environment/javascript/nodejs-on-windows)

```shell
winget install --id=CoreyButler.NVMforWindows  -e
```

Followed by another line in the same script (again, as per the docs):
```shell
nvm install latest
```

Results in:

```
PS C:\Users\User\AppData\Roaming\nvm> nvm install latest
nvm : The term 'nvm' is not recognized as the name of a cmdlet, function, script file, or operable program.
```

### Proposed technical implementation details

WinGet or Terminal should update the user environment variables in order to unblock automated installations of software

**Comment by Trenly** [comment]
@stephengillie - Area-Path also

**Comment by denelon** [comment]
Duplicate of #3067

**Comment by denelon** [comment]
Duplicate of #222

**Comment by globdug** [comment]
I had the same problem.

This solution seems to work for me.

```powershell
winget install --id=CoreyButler.NVMforWindows  -e --silent

$env:NVM_HOME = [System.Environment]::GetEnvironmentVariable("NVM_HOME", "User")
$env:NVM_SYMLINK = [System.Environment]::GetEnvironmentVariable("NVM_SYMLINK", "User")
$env:Path = [System.Environment]::GetEnvironmentVariable("Path", "Machine") + ";" + [System.Environment]::GetEnvironmentVariable("Path", "User")

nvm install 18.6.0
nvm use 18.6.0
```

**Comment by Trenly** [comment]
@denelon

**Comment by soredake** [comment]
Totally agree that PATH should be reloaded after installing package so i can use it right away, similar feature request for chocolatey https://github.com/chocolatey/choco/issues/2458

**Comment by denelon** [comment]
We're currently working on the dependency flow:
* https://github.com/microsoft/winget-cli/issues/2939

 You may also want to take a look at:
* https://github.com/microsoft/winget-cli/issues/549

We're looking at a few scenarios related to the path, and there is discussion and a possible workaround as well as some design work on how we might achieve this.

**Comment by AndreM222** [comment]
Has there been any advancements regarding this issue?

**Comment by moi15moi** [comment]
This seems to have been fixed with powershell 7.6. See: https://github.com/microsoft/winget-cli/issues/549
